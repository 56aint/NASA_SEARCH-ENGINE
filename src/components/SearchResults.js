import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchResults.css";

const SearchResults = ({ results }) => {
  return (
    <div className="SearchResults" data-testid="SearchResults">
        <img className="card-image" data-testid="card-image" src={results} alt="spaceImage" />
    </div>
  );
}


SearchResults.propTypes = {
  image: PropTypes.object,
  results: PropTypes.array,
};

export default SearchResults;
