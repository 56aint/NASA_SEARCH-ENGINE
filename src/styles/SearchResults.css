.SearchResults {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: auto;
  text-align: center;
  padding: 20px;
}
.card-image {
  width: 100%;
  height: 100%;
  object-fit: contain;
  border-radius: 30px;
}