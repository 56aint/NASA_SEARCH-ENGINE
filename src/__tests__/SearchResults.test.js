import React from 'react';
import { render } from '@testing-library/react';
import SearchResults from '../components/SearchResults';

describe("SearchResults", () => {
  
  const { asFragment } = render(<SearchResults/>);

  xit('renders correctly to match snapshot', () => {
    expect(asFragment()).toMatchSnapshot();
  });

  xit("renders the correct amount props", () => {
    const { getByTestId } = render(<SearchResults results="mockResults" />)

    expect(getByTestId("SearchResults")).toHaveClass("SearchResults");
    expect(getByTestId("card-image")).toHaveClass("SearchResults");
  });
});