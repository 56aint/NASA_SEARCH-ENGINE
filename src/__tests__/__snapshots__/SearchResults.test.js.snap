// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="SearchResults"
    data-testid="SearchResults"
  >
    <img
      alt="spaceImage"
      class="card-image"
      data-testid="card-image"
    />
  </div>
</DocumentFragment>
`;
